npc/re/instances/Sara'sMemory.txt

```diff
@@ -0,0 +1,1863 @@
+					erasequest 15003;
+					close;
+				}
+			}
+			if (questprogress(15003) == 2) {
+				if (!sarainstance) {	// 1st time reward and text
+					mes("What did you see in there?!");
+					next;
+					mes("[Leon the Adventurer]");
+					mes("Hrumpf, Sara Irene?! She's one of the 12 Valkyries. I didn't know her story was that tragic....");
+					next;
+					mes("[Leon the Adventurer]");
+					mes("It seems that you caused a huge misunderstanding between her and her father.");
+					next;
+					mes("[Leon the Adventurer]");
+					mes("I would not blame yourself for what	 occured. Or all the blood that was spilled. It seems that the\r"
+						"events that day were fated to happen no matter who was there.");
+					next;
+					mes("[Leon the Adventurer]");
+					mes("Please don't tell anyone about your time traveling adventure. We can't have anyone thinking you more\r"
+						"important then old Leon. Plus they wouldn't believe such a story anyways.");
+					next;
+					mes("[Leon the Adventurer]");
+					mes("I think that in about 20 hours after you last entered the rift will stable enough for you to transverse it again.");
+					sarainstance = 1;
+					getexp 770000, 1000000;	// reward first time
+				} else {	// 2nd time text
+					mes("What did you see in there?! Was there anything new?");
+					next;
+					select("Nothing has changed from the last time.");
+					mes("[Leon the Adventurer]");
+					mes("I am sorry to hear that... If...");
+					next;
+					mes("[Leon the Adventurer]");
+					mes("If you want to try again. I think that in about 20 hours after you last entered the rift will be stable enough\r"
+						"for you to transverse it again.");
+					next;
+					mes("[Leon the Adventurer]");
+					mes("You should rest up. Travelling Through time would make even Leon tired!");
+					getexp 550000, 550000;	// reward 2nd and following
+				}
+				erasequest 15003;
+				close;
+			}
+		}
+
+dali,138,118,0	script	Dimensional Device#dimen	PORTAL,{
+	if (!questprogress(15003)) {
+		mes("[Leon the Adventurer]");
+		mes("Before you go playing around with that. Why don't you come over here and talk to me?");
+		close;
+	}
+		.@sara_time = questprogress(15002, PLAYTIME);
+		if ( .@sara_time == 1) {
+			mes("[Leon the Adventurer]");
+			mes("I think that in about 20 hours after you last entered the rift will stable enough for you to\r"
+				"transverse it again.");
+			close;
+		} else if (.@sara_time == 2) {
+			erasequest 15002;
+			end;
+		} else if (!.@sara_time) {
+			.@party_id = getcharid(1);
+			.@p_name$ = getpartyname(.@party_id);
+			.@md_name$ = "Sara's Memory";
+			if (!instance_check_party(.@party_id)) {
+				mes("[Leon the Adventurer]");
+				mes("Before you enter you need to organize a party! Don't know how? Type /organize PARTYNAMEHERE. You\r"
+					"can use quotes to put spaces in a party name.");
+				close;
+			}
+			if (getcharid(0) == getpartyleader(.@party_id, 2))
+				.@menu1$ = "Boot up the dimensional device";
```

Comment:
maybe an additional check here
`if (has_instance("1@sara"))`
to remove the choise instead of show an error ? or it's officialy like this?